app/models/manageiq/providers/vmware/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,119 @@
+class ManageIQ::Providers::Vmware::Inventory::Parser::CloudManager < ManageIQ::Providers::Vmware::Inventory::Parser
+  def parse
+    orgs
+    vdcs
+    vapps
+    vms
+    vapp_templates
+    images
+  end
+
+  private
+
+  def orgs
+    collector.orgs
+  end
+
+  def vdcs
+    collector.vdcs.each do |vdc|
+      persister.availability_zones.find_or_build(vdc.id).assign_attributes(
+        :ems_ref => vdc.id,
```

Comment:
I think you don't need to store `ems_ref` explicitly, because you pass it's value as an argument to `find_or_build()` already. Can you please check if you can omit it? If yes, then please remove it also in all other functions below.